modules/cf-core/src/ethereum/multisig-commitment.ts

```diff
@@ -44,12 +52,42 @@ export abstract class MultisigCommitment extends EthereumCommitment {
   }
 
   public hashToSign(): string {
-    const { to, value, data, operation } = this.getTransactionDetails();
-    return keccak256(
-      solidityPack(
-        ["bytes1", "address[]", "address", "uint256", "bytes", "uint8"],
-        ["0x19", this.multisigOwners, to, value, data, operation]
-      )
+    const {
+      to,
+      value,
+      data,
+      operation,
+      domainName,
+      domainVersion,
+      chainId,
+      domainSalt,
+      transactionCount
+    } = this.getTransactionDetails();
+    const domainSeparatorHash = solidityKeccak256(
+      ["bytes32", "bytes32", "uint256", "address", "bytes32"],
+      [id(domainName), id(domainVersion), chainId, this.multisigAddress, domainSalt]
```

Comment:
A4 - Sign domain separator hash into hashed transaction details for multisig.
A3 - Fix for unsafe `encodePacked`.